spec/lib/whitehall_importer_spec.rb

```diff
@@ -101,5 +101,22 @@
       WhitehallImporter.sync(whitehall_import)
       expect(whitehall_import).to be_completed
     end
+
+    context "when the sync fails" do
+      before do
+        allow(ResyncService).to receive(:call)
+          .with(whitehall_import.document)
+          .and_raise(GdsApi::HTTPTooManyRequests.new(429, message))
+      end
+
+      let(:message) { "Ahhh too many requests" }
+
+      it "marks the import as failed due to sync issues and logs the error" do
+        WhitehallImporter.sync(whitehall_import)
+
+        expect(whitehall_import).to be_sync_failed
+        expect(whitehall_import.error_log).to eq(message)
```

Comment:
Not really a problem for now but I wonder if what we're storing in this error log isn't really sufficient without the class name of the exception 🤔

Reply:
I had a little think about this and think it's okay as we have quite descriptive abort messages and any ruby / rails exception should have a pretty clear message of went wrong without needing the actual class. 

Possibly the error messages from GdsApi may be poor but I'd hope that's not the case.

Reply:
Yeah I think it's fine for abort too as we already know it from the status.

For an unexpected exception I had a little play around and think using inspect might be the most promising approach:

```
[3] pry(#<GdsApi::TestHelpers::PublishingApi::#stub_publishing_api_has_path_reservation_for>)> error.to_s
=> "URL: http://publishing-api.dev.gov.uk/paths/foo\nResponse body:\n{\"error\":{\"code\":422,\"message\":\"Base path /foo is already reserved by foo-publisher\",\"fields\":{\"base_path\":[\"/foo is already reserved by foo-publisher\"]}}}\n\nRequest body:\n{:publishing_app=>\"bar-publisher\"}"
[4] pry(#<GdsApi::TestHelpers::PublishingApi::#stub_publishing_api_has_path_reservation_for>)> error.class.name
=> "GdsApi::HTTPUnprocessableEntity"
[5] pry(#<GdsApi::TestHelpers::PublishingApi::#stub_publishing_api_has_path_reservation_for>)> error.inspect
=> "#<GdsApi::HTTPUnprocessableEntity: URL: http://publishing-api.dev.gov.uk/paths/foo\nResponse body:\n{\"error\":{\"code\":422,\"message\":\"Base path /foo is already reserved by foo-publisher\",\"fields\":{\"base_path\":[\"/foo is already reserved by foo-publisher\"]}}}\n\nRequest body:\n{:publishing_app=>\"bar-publisher\"}>"
```

Reply:
Inspect looks like the best of all worlds so will alter the code accordingly!